frontend/apps/web/components/connections/TestConnectionBadge.tsx

```diff
@@ -2,93 +2,90 @@
 import { CheckConnectionConfigResponse } from '@neosync/sdk';
 import { ArrowTopRightIcon, CheckCircledIcon } from '@radix-ui/react-icons';
 import Link from 'next/link';
+import { ReactElement } from 'react';
 import { MdErrorOutline } from 'react-icons/md';
 import { TiWarningOutline } from 'react-icons/ti';
 
 interface TestConnectionBadgeProps {
   validationResponse: CheckConnectionConfigResponse | undefined;
-  id: string | undefined;
+  connectionId: string | undefined;
   accountName: string;
 }
 
-export default function TestConnectionBadge(props: TestConnectionBadgeProps) {
-  const { validationResponse, id, accountName } = props;
+export default function TestConnectionBadge(
+  props: TestConnectionBadgeProps
+): ReactElement {
+  const { validationResponse, connectionId, accountName } = props;
 
   return (
     <ValidationResponseBadge
       validationResponse={validationResponse}
       accountName={accountName}
-      id={id ?? ''}
+      connectionId={connectionId ?? ''}
     />
   );
 }
 
 interface ValidationResponseBadgeProps {
   validationResponse: CheckConnectionConfigResponse | undefined;
   accountName: string;
-  id: string;
+  connectionId: string;
 }
 
-function ValidationResponseBadge(props: ValidationResponseBadgeProps) {
-  const { validationResponse, accountName, id } = props;
-  const url = `/${accountName}/connections/${id}/permissions`;
+function ValidationResponseBadge(
```

Comment:
reshuffled the logic here to more clearly represent code paths. Also added check to just return null if there is no validation response